### ISSUE TITLE
AucPrMetric reports order-dependent Average Precision for tied scores

### ISSUE THREAD

**Issue by Mikyx-1** [tracel-ai/burn#5683]
## Describe the bug

`AucPrMetric` can report different Average Precision values for the same samples in a different order when prediction scores tie. In a two-sample example, it reports either 100% or 50% even though both predictions are 0.5.

Reproduced on `b7946a625c56f65e5f51d435c60cb204657141ec`, Linux x86_64, Flex.

## Reproduction

```rust
use burn_core::{
    data::dataloader::Progress,
    tensor::{Device, Tensor},
};
use burn_train::metric::{
    AucPrMetric, ConfusionStatsInput, Metric, MetricMetadata, Numeric,
};

fn main() {
    let device = Device::flex();
    let metadata = MetricMetadata {
        progress: Progress {
            items_processed: 2,
            items_total: 2,
            unit: None,
        },
        iteration: Some(0),
        lr: None,
    };
    for labels in [[[true], [false]], [[false], [true]]] {
        let input = ConfusionStatsInput::new(
            Tensor::from_floats([[0.5], [0.5]], &device),
            Tensor::from_data(labels, &device),
        );
        let mut metric = AucPrMetric::binary();
        metric.update(&input, &metadata);
        metric.compute();
        println!("{:?}: {}%", labels, metric.final_value().current());
    }
}
```

Actual results:

```text
[[true], [false]]: 100%
[[false], [true]]: 50%
```

Both should be 50%. A score threshold includes both tied samples together: precision is 1/2 when recall reaches 1. I also checked `sklearn.metrics.average_precision_score` in scikit-learn 1.9.0, which returns 0.5 for both orders.

## Cause and proposed fix

`average_precision` in `crates/burn-train/src/metric/auc_pr.rs` sorts scores, then evaluates cumulative precision at every sample rank. That treats tied scores as distinct thresholds and makes the result depend on their ordering.

Compute precision at the end of each equal-score group, using the whole group when assigning its recall contribution. This should make results invariant to sample order and agree with the documented threshold-based Average Precision definition, including macro and micro reductions.

I am preparing a focused fix with regression tests for tied groups, batch boundaries, and an independent reference calculation.
